slingshotapi/vendors/aws/s3adaptor.py

```
@@ -88,3 +90,23 @@ def _get_region():
         except requests.exceptions.ConnectionError:
             region = "eu-west-1"
         return region
+
+    def _get_regional_client_for_bucket(self, bucket):
+        """Return an S3 boto3 client in the same region as the specified bucket
+
+        # Creating a boto3 client in the target location as there are signing inconsistencies
+        # when generating cross-regional presigned urls
+
+        :param bucket: An S3 bucket name.
+        :type bucket: str
+
+        :returns: An S3 boto3 client
+        :rtype: boto3.client
+        """
+        try:
+            target_region = self.client.get_bucket_location(Bucket=bucket)['LocationConstraint']
+            upload_client = boto3.client(
+                's3', region_name=target_region)
+        except requests.exceptions.ConnectionError:
+            upload_client = self.client
+        return upload_client
```

Also, I don't think the try / except here is necessarily required as the `_get_regional_client_for_bucket()` method is being called from within a try / except (which catches botocore client exceptions) in another method. The calling method will catch the botocore exception and raise the error.
Mentioning this as the current logic will pass a client configured for the current region back to the calling method which will then try to use it sign urls. If it's a v2 client and it's signing for a v4 region the incorrect url will be passed back to the calling client which will then fail to upload as before.
We should be failing fast and returning a suitable message to the calling client which is what will happen if we raise the unhandled botocore exception back to the calling method, i.e. the signing attempt will fail, no urls will be generated, and the calling client will be immediately informed of the signing failure.

With the above in mind, you could cut the method body back to:
```
    target_region = self.client.get_bucket_location(Bucket=bucket)['LocationConstraint']
    return boto3.client('s3', region_name=target_region)
```